index.rst

```diff
@@ -42,20 +42,163 @@
+   This is because the prompt processing system uses the group ID to collate data and expects a specific value to appear that matches the value published by the ScriptQueue.
+
+As pipelines execute, they will be able to calculate data quality assessment metrics and other results of interest such as centroids.
+These can be published as telemetry to allow them to be displayed to the observer.
+This telemetry can be received by the observing script that issued the data processing request, allowing the script to fine tune its observing strategy.
+These metrics should be associated with the history of the exposure in the same way as observer comments are associated with specific observations.
+
+.. note::
+
+   Notebooks can also block waiting for the processed data events to arrive before doing bespoke commissioning visualization or further detailed analysis that is decoupled from observing.
+
+The OCPS CSC
+------------
+
+The OCPS CSC will support one command, let's say ``processData``, used by observing scripts.
+The command will need to take at least a dataId and a pipeline name, and can possibly also take configuration overrides.
+The dataId looks like a ``dict`` and would be expected to include either an obsid or a group ID and possibly a detector number.
+The CSC will publish an event when a processing request has been completed.
+It will also publish any telemetry from the pipelines themselves.
+
+The OCPS itself acts as a small interface between the OCS and a pipeline execution environment.
+The Observing Script does not have to know how pipelines are executed, all it needs to know is that named pipelines can be executed and will report when they have completed.
+
+In the interim before Gen 3 is fully deployed, the OCPS could be something that launches a command line task.
+It could also be a full Gen 3 pipeline executed by ``pipetask``.
+It is though important that the OCPS knows when processing completes and also knows whether processing succeeded.
+There also needs to be a mechanism for funneling metrics from the pipelines to the OCPS so that they can be issued as telemetry.
+
+Open Questions
+--------------
+
+There are some details that need to be considered:
+
+* If there is a delay in ingesting the previous observation into the OODS, how does the observing script know that the event that was received is for the correct observation?
+  Should it listen for the camera event indicating that the data were taken and read the obsid from that event?
+* If the OCPS is still handling the previous processing request when the new command arrives what should happen?
+  Should it return immediately with a busy status?
+  Should it queue the request?
+  Should it block until the prior processing has completed but timeout if it is taking too long?
+* Should the OCPS check that the data really are in the OODS before accepting the command?
```

Comment:
It is not clear that the OCPS can return a more meaningful error message if the data are not present than the pipeline itself, and the latter seems simpler.